### ISSUE TITLE
Custom dataType Not Working

### ISSUE THREAD

**Issue by willrstern** [js-data/js-data-schema#12]
Always possible that I'm missing something...
But I can't get custom data types working: http://codepen.io/anon/pen/VYNNQE?editors=001

```javascript
var schemator = new Schemator();

schemator.defineDataType('NaN', function (x) {
  if (isNaN(x)) {
    return null;
  } else {
    return {
      rule: 'type',
      actual: typeof x,
      expected: 'NaN'
    };
  }
});

var schema = schemator.defineSchema('mySchema', {
  notNumber: {
    type: 'NaN'
  }
});

var errors = schema.validateSync({
  notNumber: 123
});

console.log('never errors for any value?', errors);
```

**Comment by jmdobry** [comment]
Fixed in 1.2.1

**Comment by willrstern** [comment]
Thanks!
